staging/src/k8s.io/api/roundtrip_test.go

```diff
@@ -150,7 +149,9 @@ func TestRoundTripExternalTypes(t *testing.T) {
 	scheme := runtime.NewScheme()
 	codecs := serializer.NewCodecFactory(scheme)
 	for _, builder := range groups {
-		require.NoError(t, builder.AddToScheme(scheme))
+		if err := builder.AddToScheme(scheme); err != nil {
+			t.Fatal(err)
```

Comment:
Sorry, but this is a perfect example why test frameworks have a value that is not immediately obvious: if this fails now, it only dumps the error text into the `go test` output.

Depending on the error, it will not be obvious that this is the reason for the test failure.

At the very least, use `t.Fatalf` with some boiler plate text that will make the log output stand out more, lie `t.Fatalf("FATAL: unexpect error: %v", err)`.

`t.Fatal` itself doesn't add any additional information. It's  the same as:
```
t.Log(err)
t.FailNow()
```

I'm not against removing `require.NoError`, but then lets think a bit more about what we want to replace it with.

Reply:
> Depending on the error, it will not be obvious that this is the reason for the test failure.

agree on the message logging, but adding dependencies just for that does not pay off

updated